22_pipeslab/pipeslab.txt

```diff
@@ -3,19 +3,42 @@ For each of the following, type a single command line that performs:
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
   Answer: 
 
+  tocode@tocode-VirtualBox:~$ ls -alF | grep tocode | cut -c4 | grep x | wc | cut -d' ' -f6
+  20
+  grep x is not must
+  
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
   Answer: 
-
+ 
+  ls -alF | grep tocode | grep x..x..x | wc | cut -d' ' -f6
+  
+  
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
   Answer: 
 
+  ls -alF | grep tocode | grep Jan..2 | wc | cut -d' ' -f7
```

Comment:
The idea here is that the date should not be hard coded.. once we 
play enough with the date command, we can see that:
```
date "+%b" # will print the current month
date "+%e" # will print the day  (1-31), space padded
```
Now we just need to feed their input  into grep and count lines, let's use $():
```
echo  "$(date +%b) $(date +%e)" # will print "ינו  7" (without quotes of course..)
ls -l | grep "$(date +%b) $(date +%e)" | wc -l
```
Now as a last touch, the fact the we check for a single white space kinda bothers me, let's fix it:  
```ls -l | grep "$(date +%b)\s* $(date +%e)" | wc -l``` 
The pattern ```\s*``` means 0 or more spaces, and we follow it with a white-space.

There are many ways to solve this, this is just one of them :)